wapi/session.py

```diff
@@ -431,16 +431,16 @@ def _build_curve(self, metadata):
             return c
         raise CurveException('Unknown curve type ({})'.format(metadata['curve_type']))
 
-    def _get_auth_header_with_retry(self, databytes, retries=RETRY_COUNT):
+    def _get_auth_header_with_retry(self, retries=RETRY_COUNT):
```

Comment:
Do we want to keep the retry in the session object? Alternative: move it into auth module. Advantage would be that everything is closer together and easier to read what happens. Assumption: One reason why the previous PR introducing the `_get_auth_header_with_retry` method passed was that `self.auth` was seen as a black box.

Reply:
Yes it makes sense to me to move the retry logic to the auth module, because like you wrote, easier to read and reason about. What do you think @stigalvestad, should I implement that in this PR?

Reply:
I agree that it makes sense to move it into auth. I would recommend to continue do retries as an opt-in feature for now, to reduce the amount of changes we introduce in one release though.